tools/internal_ci/linux/grpc_bazel_build_in_docker.sh

```diff
@@ -26,3 +26,7 @@ ${name}')
 cd /var/local/git/grpc
 
 bazel build :all //test/... //examples/...
+# This tests that builds with --define=grpc_no_xds=true work. It can be made a
+# standalone script if needed.
+bazel build //test/cpp/end2end:end2end_test --define=grpc_no_xds=true
```

Comment:
Some other things that might be useful to check:
- Compare the binary sizes of end2end_test when built both with and without xDS, and make sure the latter is smaller.
- Try building xds_end2end_test with --define=grpc_no_xds=true and make sure it fails.

Reply:
Added